qiskit/utils/run_circuits.py

```diff
@@ -308,7 +308,7 @@ def run_qobj(
                 logger.info("Running %s-th qobj, job id: %s", idx, job_id)
                 # try to get result if possible
                 while True:
-                    job_status = _safe_get_job_status(job, job_id, max_job_retries)
+                    job_status = _safe_get_job_status(job, job_id, max_job_retries, qjob_config["wait"])
```

Comment:
Looking inside `_safe_get_job_status` the `sleep` is only used if the test for status fails, then it is tested again a `max_job_retries` times until the status check passes. If it doesn't pass after the number of tries, it will raise an exception and all will fail. The methods that call it, do other tests with the status that was obtained. If the status is not in one of the `JOB_FINAL_STATES` it will do another `sleep` and try again. So the 2 sleeps are for different situations. removing  `_safe_get_job_status` means adding its loop logic to get status and check for failure inside the other 2 places.

Reply:
Ah, yes that makes sense. Thanks for checking!